.eslintrc.js

```diff
@@ -106,9 +106,40 @@ module.exports = {
 		// i18n-calypso translate triggers false failures
 		'jsx-a11y/anchor-has-content': 0,
 
-		// error if any module depends on the data-observe mixin, which is deprecated
-		'no-restricted-imports': [ 2, 'lib/mixins/data-observe' ],
-		'no-restricted-modules': [ 2, 'lib/mixins/data-observe' ],
+		'no-restricted-imports': [
+			2,
+			{
+				paths: [
+					// Error if any module depends on the data-observe mixin, which is deprecated.
+					'lib/mixins/data-observe',
+					// Prevent naked import of gridicons module. Use 'components/gridicon' instead.
+					{
+						name: 'gridicons',
+						message: "Please use 'components/gridicon' instead.",
+					},
+					// Prevent importing Redux's combineReducers.
+					{
+						name: 'redux',
+						importNames: [ 'combineReducers' ],
+						message: "`combineReducers` should be imported from 'state/utils', not 'redux'.",
+					},
+				],
+			},
+		],
+		'no-restricted-modules': [
+			2,
+			{
+				paths: [
+					// Error if any module depends on the data-observe mixin, which is deprecated.
+					'lib/mixins/data-observe',
+					// Prevent naked import of gridicons module. Use 'components/gridicon' instead.
+					{
+						name: 'gridicons',
+						message: "Please use 'components/gridicon' instead.",
+					},
+				],
```

Comment:
Note that we can't check for Redux `combineReducers` in CommonJS, as there's no concept of a non-default import there.